### PR TITLE
Add (i)CFG node type annotations + bug fix

### PR DESCRIPTION
**Pull request by Rot127**
**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [x] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [x] I've documented or updated the documentation of every function and struct this PR changes. If not so I've explained why.
- [x] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the [rizin book](https://github.com/rizinorg/book) with the relevant information (if needed)

**Detailed description**

Fixes the bug that the buffer was not read again, before nodes to `RzAnalysisOp.jump/fail` were added.

Additionally, and for testing specifically this, annotations are added to the CFG/iCFG nodes.

New annotations:

- None: `○` no-utf8: `.`
- Cond: `⤹` no-utf8: `c`
- Entry: `↓` no-utf8: `e`
- Return: `↑` no-utf8: `r`
- Call: `⇢` no-utf8: `C`
- Exit: `⭳` no-utf8: `E`

Exit is not yet tested, because it is not implemented yet. Exit nodes are nodes which exit the program abnormally, although the procedure should normally return (e.g. call to `abort()`).

```asm
            ;-- ignore_call:
            0x080000c3      push  rbp
            0x080000c4      mov   rbp, rsp
            0x080000c7      call  reloc.target.rand                    ; RELOC 32 rand
            0x080000cc      nop
            0x080000cd      pop   rbp
            0x080000ce      ret
```

```
>rizin -Qc 's sym.ignore_call
agF
' test/bins/elf/analysis/x86_cfg_test
┌─────────────────┐
│  0x80000c3 ↓    │
└─────────────────┘
    v
    │
    │
┌─────────────────┐
│  0x80000c4 ○    │
└─────────────────┘
    v
    │
    │
┌─────────────────┐
│  0x80000c7 ⇢    │
└─────────────────┘
    v
    │
    │
┌─────────────────┐
│  0x80000cc ○    │
└─────────────────┘
    v
    │
    │
┌─────────────────┐
│  0x80000cd ○    │
└─────────────────┘
    v
    │
    │
┌─────────────────┐
│  0x80000ce ↑    │
└─────────────────┘

> rizin -Qc 's sym.ignore_call
e scr.utf8=0 ; agF
' test/bins/elf/analysis/x86_cfg_test
.---------------.
|  0x80000c3 e  |
`---------------'
    v
    |
    |
.---------------.
|  0x80000c4 .  |
`---------------'
    v
    |
    |
.---------------.
|  0x80000c7 C  |
`---------------'
    v
    |
    |
.---------------.
|  0x80000cc .  |
`---------------'
    v
    |
    |
.---------------.
|  0x80000cd .  |
`---------------'
    v
    |
    |
.---------------.
|  0x80000ce r  |
`---------------'

```

**Test plan**

All green + added

**Closing issues**

...

### CODE REVIEW

File: librz/core/cagraph.c [hunk 1]
```diff
@@ -268,7 +268,7 @@ RZ_IPI bool rz_core_agraph_apply(RzCore *core, RzGraph /*<RzGraphNodeInfo *>*/ *
 	if (!(core && core->graph && graph)) {
 		return false;
 	}
-	if (!create_agraph_from_graph_at(core->graph, graph, false)) {
+	if (!create_agraph_from_graph_at(core->graph, graph, false, rz_config_get_i(core->config, "scr.utf8"))) {
```
Comment (XVilka) [inline librz/core/cagraph.c]: `rz_config_get_b()`

File: librz/core/agraph.c [hunk 2]
```diff
@@ -3740,16 +3740,20 @@ RZ_API RZ_BORROW RzANode *rz_agraph_add_node_from_node_info(RZ_NONNULL const RzA
 		}
 		an->offset = info->def.offset;
 		break;
-	case RZ_GRAPH_NODE_TYPE_CFG:
-		rz_strf(title, "0x%" PFMT64x, info->cfg.address);
-		an = rz_agraph_add_node(g, title, "");
+	case RZ_GRAPH_NODE_TYPE_CFG: {
+		char *cfg_title = rz_str_newf("0x%" PFMT64x, info->cfg.address);
+		rz_str_append(cfg_title, rz_graph_get_node_subtype_annotation(info->subtype, utf8));
```
Comment (XVilka) [inline librz/core/agraph.c]: You could use `rz_str_appendf()` directly

File: librz/core/agraph.c [hunk 3]
```diff
@@ -3725,10 +3725,10 @@ RZ_API void rz_agraph_set_title(RzAGraph *g, const char *title) {
 /**
  * \brief Convert a RzGraphNodeInfo \p info to RzANode and add to \p g.
```
Comment (XVilka) [inline librz/core/agraph.c:3726]: Update Doxygen then too

File: librz/core/agraph.c [hunk 4]
```diff
@@ -4962,7 +4966,7 @@ RZ_IPI int rz_core_visual_graph(RzCore *core, RzAGraph *g, RzAnalysisFunction *_
  * \brief Create RzAGraph from generic RzGraph with RzGraphNodeInfo as node data at \p ag from \p g
  * \return Success
```
Comment (XVilka) [inline librz/core/agraph.c]: Same, update Doxygen

File: librz/core/agraph.c [hunk 5]
```diff
@@ -5021,14 +5025,14 @@ RZ_API bool create_agraph_from_graph_at(RZ_NONNULL RzAGraph *ag, RZ_NONNULL cons
  * \param graph <RzGraphNodeInfo>
  * \return RzAGraph* NULL if failure
```
Comment (XVilka) [inline librz/core/agraph.c]: Same

File: librz/util/graph_drawable.c [hunk 6]
```diff
@@ -5,6 +5,30 @@
 #include <rz_core.h>
 #include <rz_util/rz_graph_drawable.h>
 
+RZ_API RZ_OWN char *rz_graph_get_node_subtype_annotation(RzGraphNodeSubType subtype, bool utf8) {
```
Comment (XVilka) [inline librz/util/graph_drawable.c:16]: Doxygen

File: test/db/cmd/cmd_graph [hunk 7]
```diff
@@ -636,39 +636,39 @@ s sym.ignore_call
 agF
 EOF
 EXPECT=<<EOF
-.-------------.
-|  0x80000c3  |
-`-------------'
+.---------------.
+|  0x80000c3 e  |
```
Comment (XVilka) [inline test/db/cmd/cmd_graph]: I wonder if it's better to put letters into braces e.g. `(e)` or `{e}`, so they will be more separate from the number, especially those that are in hex range, e.g. `e`.

**Review by XVilka — CHANGES_REQUESTED** [review]

**Comment by Rot127** [comment]
No idea what is wrong with the Windows test. Spin up a Windows VM tomorrow.

**Review by XVilka — CHANGES_REQUESTED** [review]
For some reason it fails tests on Windows: https://ci.appveyor.com/project/rizinorg/rizin/builds/49184102/job/s27n8bx7hn9ix46e#L4814

**Comment by Rot127** [comment]
~~Blind fix, though I think it was the leak.~~ rz_str_append() frees the pointer for realloc.

**Review by XVilka — APPROVED** [review]

**Pull request merged**